### PR TITLE
Issue #12000 - use URIUtil::toURI instead of URI::create in MavenWebAppContext

### LINKED ISSUE #12000: Cannot use Paths that have spaces with `jetty-ee10-maven-plugin`
**Jetty version(s)**
12.0.10

**Jetty Environment**
Any (just tested in ee10)

**Java version/vendor** `(use: java -version)`
Any

**OS type/version**
Any (tested in Unix)

**Description**
Create a webapp project in a path on unix that has a space.

```
/home/user/code/Project Foo/Admin WebApp/pom.xml
```

Now try to specify the `<resourceBases>` section using either `${basedir}` or `${project.build.directory}` in the `jetty-ee10-maven-plugin`
It will fail as the URI.create() will not like the raw space.

https://github.com/jetty/jetty.project/blob/05a0498627d3cd2a73a0bdfbb4550d7d2af9f67c/jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/MavenWebAppContext.java#L225-L243

### PR DESCRIPTION
**Pull request by joakime**
Restore `URIUtil.toURI(String)`, properly javadoc it, and use it where end-user provided Strings are given.
Include maven-plugin IT test that replicates reported issue.

Fixes #12000

### CODE REVIEW

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java [hunk 1]
```diff
@@ -1892,24 +1892,27 @@ else if (scheme.equalsIgnoreCase("file"))
     }
 
     /**
-     * <p>Convert a String into a URI suitable for use as a Resource.</p>
+     * <p>Convert a String into a URI in a sane way.</p>
```
Comment (gregw) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java:1895]: you need to say it better than "in a sane way". what does that mean?

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java [hunk 2]
```diff
@@ -1892,24 +1892,27 @@ else if (scheme.equalsIgnoreCase("file"))
...
      *
-     * @param resource If the string starts with one of the ALLOWED_SCHEMES, then it is assumed to be a
-     * representation of a {@link URI}, otherwise it is treated as a {@link Path}.
-     * @return The {@link URI} form of the resource.
-     * @deprecated This method is currently resolving relative paths against the current directory, which is a mechanism
-     * that should be implemented by a {@link ResourceFactory}.   All calls to this method need to be reviewed.
+     * <p>
+     *     This exits to take an end user provided String and make a usable URI out of it.
```
Comment (gregw) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java:1898]: No indent.
s/exits/exists/

File: jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java [hunk 3]
```diff
@@ -1892,24 +1892,27 @@ else if (scheme.equalsIgnoreCase("file"))
...
+     *     It is capable of dealing with paths that have spaces, windows slashes, windows UNC references,
+     *     relative paths, absolute paths, and much more.
```
Comment (gregw) [inline jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java:1900]: This still feels like a job of a `ResourceFactory`.  Why is this needed?

File: jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/MavenWebAppContext.java [hunk 4]
```diff
@@ -231,7 +231,7 @@ public void setResourceBases(String[] resourceBases)
             // This is a user provided list of configurations.
             // We have to assume that mounting can happen.
             List<URI> uris = Stream.of(resourceBases)
-                .map(URI::create)
+                .map(URIUtil::toURI)
```
Comment (gregw) [inline jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/MavenWebAppContext.java:234]: ResourceFactory is the class we should be using to convert user provided strings. Especially as what we want is a CompositeResource at the end.

**Review by gregw — CHANGES_REQUESTED** [review]
I'm not in favour of resurrecting this method.  Use a ResourceFactory.  If that is difficult, then improve ResourceFactory.   Having a resource-factory-lite method handling user input is asking for trouble.    What if they give us a jar:file resource?

**Comment by janbartel** [comment]
@joakime I made PR #12003 to fix the problem with the jetty maven plugin.

**Comment by janbartel** [comment]
@joakime oops, when I committed my PR #12003 I think it closed the associated issue.

**Comment by janbartel** [comment]
@joakime I think we've addressed the issue with PR #12003 so I'll close this PR. If there's still work to do, we need a new issue for it.

**Pull request closed without merging**